src/EnergyPlus/RoomAirModelManager.cc

```diff
@@ -1052,139 +967,108 @@ namespace RoomAirModelManager {
-                    state.dataRoomAirModelMgr->TypeNum = state.afn->AirflowNetworkCompData(state.dataRoomAirModelMgr->CompNum).TypeNum;
-                    if (state.afn->AirflowNetworkCompData(state.dataRoomAirModelMgr->CompNum).CompTypeNum == AirflowNetwork::iComponentTypeNum::SCR) {
-                        if (state.afn->MultizoneSurfaceCrackData(state.dataRoomAirModelMgr->TypeNum).exponent != 0.50) {
-                            state.dataRoomAirMod->AirModel(ThisZone).AirModelType = DataRoomAirModel::RoomAirModel::Mixing;
-                            ShowWarningError(state, format("Problem with {} = {}", cCurrentModuleObject, state.dataIPShortCut->cAlphaArgs(1)));
-                            ShowWarningError(state, format("Roomair model will not be applied for Zone={}.", state.dataIPShortCut->cAlphaArgs(1)));
+            for (int iLink = 1; iLink <= state.afn->NumOfLinksMultiZone; ++iLink) {
+                auto const &mzSurfaceData = state.afn->MultizoneSurfaceData(iLink);
+                int nodeNum1 = mzSurfaceData.NodeNums[0];
+                int nodeNum2 = mzSurfaceData.NodeNums[1];
+                if (state.dataSurface->Surface(mzSurfaceData.SurfNum).Zone == zoneCV.ZonePtr ||
+                    (state.afn->AirflowNetworkNodeData(nodeNum2).EPlusZoneNum == zoneCV.ZonePtr &&
+                     state.afn->AirflowNetworkNodeData(nodeNum1).EPlusZoneNum > 0) ||
+                    (state.afn->AirflowNetworkNodeData(nodeNum2).EPlusZoneNum > 0 &&
+                     state.afn->AirflowNetworkNodeData(nodeNum1).EPlusZoneNum == zoneCV.ZonePtr)) {
+                    int compNum = state.afn->AirflowNetworkLinkageData(iLink).CompNum;
+                    int typeNum = state.afn->AirflowNetworkCompData(compNum).TypeNum;
+                    if (state.afn->AirflowNetworkCompData(compNum).CompTypeNum == AirflowNetwork::iComponentTypeNum::SCR) {
+                        if (state.afn->MultizoneSurfaceCrackData(typeNum).exponent != 0.50) {
+                            state.dataRoomAir->AirModel(zoneCV.ZonePtr).AirModel = RoomAirModel::Mixing;
+                            ShowWarningError(state, format("Problem with {} = {}", ipsc->cCurrentModuleObject, ipsc->cAlphaArgs(1)));
+                            ShowWarningError(state, format("Roomair model will not be applied for Zone={}.", ipsc->cAlphaArgs(1)));
                             ShowContinueError(
                                 state,
                                 format("AirflowNetwrok:Multizone:Surface crack object must have an air flow coefficient = 0.5, value was={:.2R}",
-                                       state.afn->MultizoneSurfaceCrackData(state.dataRoomAirModelMgr->TypeNum).exponent));
+                                       state.afn->MultizoneSurfaceCrackData(typeNum).exponent));
                         }
                     }
-                }
-            }
-        }
+                } // if 
```

Comment:
I like to add these annotations. They help me remember where I am in the current nesting level.